impl/src/test/java/org/ehcache/docs/ConfigurationDerivation.java

```diff
@@ -0,0 +1,224 @@
+      .build();
+
+    //tag::withServiceCreation[]
+    Configuration withBoundedThreads = configuration.derive()
+      .withService(new PooledExecutionServiceConfiguration()
+        .addDefaultPool("default", 1, 16))
+      .build();
+    //end::withServiceCreation[]
+
+    Assert.assertThat(configuration.getServiceCreationConfigurations(), IsNot.not(IsCollectionContaining.hasItem(IsInstanceOf.instanceOf(PooledExecutionServiceConfiguration.class))));
+    PooledExecutionServiceConfiguration serviceCreationConfiguration = ServiceUtils.findSingletonAmongst(PooledExecutionServiceConfiguration.class, withBoundedThreads.getServiceCreationConfigurations());
+    Assert.assertThat(serviceCreationConfiguration.getDefaultPoolAlias(), Is.is("default"));
+    Assert.assertThat(serviceCreationConfiguration.getPoolConfigurations().keySet(), IsIterableContainingInAnyOrder.containsInAnyOrder("default"));
+    PooledExecutionServiceConfiguration.PoolConfiguration pool = serviceCreationConfiguration.getPoolConfigurations().get("default");
+    Assert.assertThat(pool.minSize(), Is.is(1));
+    Assert.assertThat(pool.maxSize(), Is.is(16));
+  }
+
+  @Test
+  public void updateServiceCreation() {
+    @SuppressWarnings("unchecked")
+    Configuration configuration = ConfigurationBuilder.newConfigurationBuilder()
+      .withService(new DefaultPersistenceConfiguration(new File("temp")))
+      .withCache("cache", CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class, ResourcePoolsBuilder.heap(10)))
+      .build();
+
+    //tag::updateServiceCreation[]
+    Configuration withUpdatedPersistence = configuration.derive()
+      .updateServices(DefaultPersistenceConfiguration.class,
+        existing -> new File("/var/persistence/path"))
+      .build();
+    //end::updateServiceCreation[]
+
+    DefaultPersistenceConfiguration initialPersistenceConfiguration = ServiceUtils.findSingletonAmongst(DefaultPersistenceConfiguration.class, configuration.getServiceCreationConfigurations());
+    Assert.assertThat(initialPersistenceConfiguration.getRootDirectory(), Is.is(new File("temp")));
+
+    DefaultPersistenceConfiguration revisedPersistenceConfiguration = ServiceUtils.findSingletonAmongst(DefaultPersistenceConfiguration.class, withUpdatedPersistence.getServiceCreationConfigurations());
+    Assert.assertThat(revisedPersistenceConfiguration.getRootDirectory(), Is.is(new File("/var/persistence/path")));
+  }
+
+  @Test
+  public void withService() {
+    Configuration configuration = ConfigurationBuilder.newConfigurationBuilder()
+      .withCache("cache", CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class, ResourcePoolsBuilder.heap(10)))
+      .build();
+
+    //tag::withService[]
+    Configuration withThrowingStrategy = configuration.derive()
+      .updateCache("cache", existing -> existing.withService(new DefaultResilienceStrategyConfiguration(new ThrowingResilienceStrategy<>())))
```

Comment:
Why didn't you push the `new ...` onto the next line to prevent an odd wrap?

Reply:
Fixed.